NaradaBotDemo/NaradaBotDemo/ProductViewController.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  ProductViewController.swift
+//  NaradaBotDemo
+//
+//  Created by Margareta Kusan on 31/08/2017.
+//  Copyright © 2017 Margareta Kusan. All rights reserved.
+//
+
+import UIKit
+
+class ProductViewController: UIViewController {
+    
+    //MARK: - Stored properties
+    let mediaContainer: UIView = {
+        let mediaContainer = UIView()
+        mediaContainer.backgroundColor = UIColor.clear
+        mediaContainer.translatesAutoresizingMaskIntoConstraints = false
+        return mediaContainer
+    }()
+    
+    let titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = UIColor.darkGray
+        label.backgroundColor = UIColor.clear
+        label.font = UIFont.systemFont(ofSize: 20)
+        return label
+    }()
+    
+    let subtitleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = UIColor.darkGray
+        label.backgroundColor = UIColor.clear
+        label.font = UIFont.systemFont(ofSize: 18)
+        return label
+    }()
+    
+    let imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.backgroundColor = UIColor.clear
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.contentMode = UIViewContentMode.scaleToFill
+        return imageView
+    }()
+    
+    let extraInfo: UITextView = {
+        let extraInfo = UITextView()
+        extraInfo.translatesAutoresizingMaskIntoConstraints = false
+        extraInfo.backgroundColor = UIColor.clear
+        extraInfo.textColor = UIColor.lightGray
+        extraInfo.font = UIFont.italicSystemFont(ofSize: 16)
+        extraInfo.isUserInteractionEnabled = false
+        return extraInfo
+    }()
+    
+    //MARK: - ViewController lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        self.view.backgroundColor = UIColor.white
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        //self.navigationController?.popViewController(animated: true)
+    }
+    
+    //MARK: - Public API
+    func fillTheView(title: String, subtitle: String, image: UIImage, littleText: String) {
+        titleLabel.text = title
+        subtitleLabel.text = subtitle
+        imageView.image = image
+        extraInfo.text = littleText
+        setupView()
+    }
+    
+    //MARK: - Private API
+    private func setupView() {
```

Comment:
this is layout code, should be inside viewDidLoad, not every time fillTheView is called